es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
SIMD is a ton of new primitive types, and Value Types will be a bunch of new ones as well. That's my concern - I'd rather not have to update es6-shim to make these new additions work.